File: product-service/src/main/java/com/ecommerce/productservice/client/member/MemberClient.java
package com.ecommerce.productservice.client.member;

import com.ecommerce.productservice.client.member.dto.SellerResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "member-service", fallbackFactory = MemberClientFallbackFactory.class)
public interface MemberClient {

    @GetMapping("/api/v1/member/{id}")
    SellerResponse getSeller(@PathVariable Long id);
}

File: review-service/src/main/kotlin/com/ecommerce/reviewservice/client/member/MemberClient.kt
package com.ecommerce.reviewservice.client.member

import com.ecommerce.reviewservice.client.member.dto.MemberInfoResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient("member-service", fallbackFactory = MemberClientFallbackFactory::class)
interface MemberClient {

    @GetMapping("/api/v1/member/infos")
    fun getMemberInfos(@RequestParam ids: List<Long>): List<MemberInfoResponse>

}